### PR TITLE
Allow plot tooltip to show extra columns

### PR DESCRIPTION
**Pull request by aliabid94**
If `tooltip="all"`, will show all columns for a datapoint, not just the axis column names
Screenshot 2024-10-22 at 11 52 35 AM

### CODE REVIEW

File: js/nativeplot/Index.svelte [hunk 1]
```diff
@@ -114,19 +115,29 @@
...
-			const obj = {
-				[x]: row[x_index],
-				[y]: row[y_index]
-			};
-			if (color && color_index !== null) {
-				obj[color] = row[color_index];
-			}
-			return obj;
-		});
+		if (tooltip == "all") {
```
Comment (aliabid94) [inline js/nativeplot/Index.svelte:118]: can't filter out "irrelevant" columns if tooltip == "all"

File: gradio/components/native_plot.py [hunk 2]
```diff
@@ -94,6 +95,7 @@ def __init__(
             x_axis_labels_visible: Whether the x-axis labels should be visible. Can be hidden when many x-axis labels are present.
             caption: The (optional) caption to display below the plot.
             sort: The sorting order of the x values, if x column is type string/category. Can be "x", "y", "-x", "-y", or list of strings that represent the order of the categories.
+            tooltip: The tooltip to display when hovering on a point. "axis" shows the values for the axis columns, "all" shows all column values, and "none" shows no tooltips.
```
Comment (abidlabs) [inline gradio/components/native_plot.py:98]: I think a better api would be to just let it take a list of column names. Often, dataframes will have a large number of irrelevant columns and I'm sure we'll get the request to be able to select specific columns to show in the tooltip

Comment (aliabid94) [reply gradio/components/native_plot.py]: Ok added support for passing list of column names

File: gradio/components/native_plot.py [hunk 3]
```diff
@@ -94,6 +95,7 @@ def __init__(
             x_axis_labels_visible: Whether the x-axis labels should be visible. Can be hidden when many x-axis labels are present.
             caption: The (optional) caption to display below the plot.
             sort: The sorting order of the x values, if x column is type string/category. Can be "x", "y", "-x", "-y", or list of strings that represent the order of the categories.
+            tooltip: The tooltip to display when hovering on a point. "axis" shows the values for the axis columns, "all" shows all column values, and "none" shows no tooltips. Can also provide a list of strings representing columns to show in the tooltip.
```
Comment (abidlabs) [inline gradio/components/native_plot.py:98]: Is it intended behavior that the axes are always shown in the tooltip? If so let's mention it here in the docstring

**Review by abidlabs — APPROVED** [review]
lgtm @aliabid94. Just noting that the axes are always shown in the tooltip, even if not explicitly listed in the list of columns e.g. see this example:

```py
import pandas as pd
import numpy as np
import gradio as gr

# Generate random data
np.random.seed(42)
dates = pd.date_range(start='2024-01-01', periods=30, freq='D')
data = {
    'Date': dates,
    'Value1': np.random.normal(100, 15, 30).cumsum(),
    'Value2': np.random.normal(50, 10, 30).cumsum()
}
df = pd.DataFrame(data)

with gr.Blocks() as demo:
    plot = gr.LinePlot(
        df,
        x="Date",
        y="Value1",
        title="Random Time Series Data",
        tooltip=["Value2"],
    )

demo.launch()
```

If this is intended behavior, let's make a note of it in the docstring

**Comment by aliabid94** [comment]
Yep its intended, added to docstring

**Comment by abidlabs** [comment]
Merging in for the release!

**Pull request merged**